addons/website/tests/test_website_visitor.py

```diff
@@ -43,14 +46,54 @@ def test_create_visitor_on_tracked_page(self):
-        self.assertEqual(len(Track.search([])), 1, "No track should be created after visiting another tracked view before 30 min")
+    def test_create_visitor_on_tracked_page(self):
+        self.assertEqual(len(self.Visitor.search([])), 0, "No visitor at the moment")
+        self.assertEqual(len(self.Track.search([])), 0, "No track at the moment")
+        self.url_open(self.untracked_view.url)
+        self.assertEqual(len(self.Visitor.search([])), 0, "No visitor created after visiting an untracked view")
+        self.assertEqual(len(self.Track.search([])), 0, "No track created after visiting an untracked view")
+        self.url_open(self.tracked_view.url)
+        self.assertEqual(len(self.Visitor.search([])), 1, "A visitor should be created after visiting a tracked view")
+        self.assertEqual(len(self.Track.search([])), 1, "A track should be created after visiting a tracked view")
+        self.url_open(self.tracked_view.url)
+        self.assertEqual(len(self.Visitor.search([])), 1, "No visitor should be created after visiting another tracked view")
+        self.assertEqual(len(self.Track.search([])), 1, "No track should be created after visiting another tracked view before 30 min")
+
+    def test_long_period_inactivity(self):
+        # link visitor to partner
+        old_visitor = self.Visitor.create({
+            'lang_id': self.env.ref('base.lang_en').id,
+            'country_id': self.env.ref('base.be').id,
+            'website_id': 1,
+        })
+        partner_demo = self.env.ref('base.partner_demo')
+        old_visitor.partner_id = partner_demo.id
+        partner_demo.visitor_ids = [(6, 0, [old_visitor.id])]  # TODO DBE : To remove in Master (13.1) after visitor_ids field declaration master fix
+        self.assertEqual(partner_demo.visitor_ids.id, old_visitor.id, "The partner visitor should be set correctly.")
+
+        # archive old visitor
+        old_visitor.last_connection_datetime = datetime.now() - timedelta(days=8)
+        self.Visitor._cron_archive_visitors()
+        self.assertEqual(old_visitor.active, False, "The visitor should be archived after one week of inactivity")
+
+        # reconnect with new visitor.
+        self.url_open(self.tracked_view.url)
+        new_visitor = self.Visitor.search([('id', '!=', old_visitor.id)])
+        self.assertEqual(len(new_visitor), 1, "A visitor should be created after visiting a tracked view")
+        self.assertEqual(len(self.Track.search([])), 1, "A track should be created after visiting a tracked view")
+
+        # override the get_visitor_from_request to mock that is new_visitor that authenticates
+        self.target_visitor = new_visitor
+        def get_visitor_from_request(self_mock, force_create=False):
+            if self.target_visitor:
+                return self.target_visitor
+            else:
+                return self.Visitor._create_visitor() if force_create else self.Visitor
```

Comment:
As you define `self.target_visitor` just above, can you reach the else case?

Reply:
ah indeed, oopsy (I defined it in the setup originally)